pkg/image/definition.go

```diff
@@ -11,12 +11,32 @@ const (
 	TypeRAW = "raw"
 )
 
+const (
+	ArchTypeX86 Arch = "x86_64"
+	ArchTypeARM Arch = "aarch64"
+)
+
 type Definition struct {
 	APIVersion      string          `yaml:"apiVersion"`
+	Arch            Arch            `yaml:"arch"`
 	Image           Image           `yaml:"image"`
 	OperatingSystem OperatingSystem `yaml:"operatingSystem"`
 }
 
+type Arch string
+
+func (a Arch) Short() string {
+	switch a {
+	case ArchTypeX86:
+		return "amd64"
+	case ArchTypeARM:
+		return "arm64"
+	default:
+		message := fmt.Sprintf("unknown arch: %s", a)
+		panic(message)
```

Comment:
This may be a Go convention (meaning the `Short()` method in general), but it feels like having a potential `panic()` in the middle of our application is a bad thing. I'm sure I'm missing something, but can you explain why both why this would have a panic included, and where it's intended to be used (I don't see any current usages)?

Reply:
It makes sense for the user to input the `x86_64` and `aarch64` values as I believe these are the most "common" ones to differentiate the instruction sets. However, there are also a lot of places where `amd64` and `arm64` are used in their places instead so I added this `Short()` function. 

Frankly speaking, I don't think it's the best way to name it but I also didn't find anything else suitable. I'm open to ideas though.

And lastly, this is going to be used for the Kubernetes RKE2 and K3S artefacts. Those include the "shortened" versions e.g. `rke2.linux-amd64`, `rke2-images-core.linux-arm64.tar.zst`, etc.

Having the `panic()` here is fine since:
- our validation will ensure that unsupported `arch` is not possible
- `Short()` can easily be used without having to handle an error case
- `recover()` is set in place in order to gracefully recover from the unexpected case of this or anything else running into a runtime error

Reply:
Ahhh, ok, that makes sense given it's usage. I'd suggest a better name but I can't think of one :)

I agree that the panic is preventable, so if you're comfortable with it, I am too.

Reply:
Yes, I'd say this is indeed one of the few cases where a `panic()` is more convenient to have. We basically want to limit the `Arch` type to an enum of the two supported values `x86_64` and `aarch64` and their _shorter_ representations.